azavea/raster-foundry
# Add external S3 bucket target to export UI

## Overview

This PR adds front-end support for specifying an external (to RF) S3 bucket for exports.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

With export target enabled:
<img width="550" alt="screen shot 2017-06-22 at 4 33 29 pm" src="https://user-images.githubusercontent.com/2442245/27454649-a36e9b68-5768-11e7-8f97-7edec0c3737c.png">

Without a valid target URI
<img width="555" alt="screen shot 2017-06-22 at 4 33 36 pm" src="https://user-images.githubusercontent.com/2442245/27454650-a36ec156-5768-11e7-8752-c9b2c1228fe6.png">

With a 'valid' target URI
<img width="542" alt="screen shot 2017-06-22 at 4 33 54 pm" src="https://user-images.githubusercontent.com/2442245/27454648-a36e2200-5768-11e7-8cb9-2f6ce2969f9d.png">

## Testing Instructions

 * With the dev-tools open so you can monitor requests, create an export for a project (you don't need to actually have airflow running if you don't want) with the 'default' export target.
 * Verify that the request body does not specifiy a `source` in the `exportOptions`
 * Do another export, but this time select 'S3 Bucket' and specify a URI (we check that it is a 'valid' url, but that is currently the extent of our validation)
 * Verify that the request body specifies a `source` that matches the URI you entered in `exportOptions`

Closes #2019
